sql/insert.go

```diff
@@ -60,33 +61,55 @@ func (p *planner) Insert(n *parser.Insert) (planNode, error) {
 		return nil, err
 	}
 
+	primaryIndex := tableDesc.Indexes[0]
+	primaryIndexKeyPrefix := encodeIndexKeyPrefix(tableDesc.ID, primaryIndex.ID)
+
 	b := client.Batch{}
+
 	for rows.Next() {
 		values := rows.Values()
 		if len(values) != len(cols) {
 			return nil, fmt.Errorf("invalid values for columns: %d != %d", len(values), len(cols))
 		}
-		indexKey := encodeIndexKeyPrefix(desc.ID, desc.Indexes[0].ID)
-		primaryKey, err := encodeIndexKey(desc.Indexes[0], colMap, values, indexKey)
+
+		primaryIndexKeySuffix, err := encodeIndexKey(primaryIndex, colMap, values, nil)
 		if err != nil {
 			return nil, err
 		}
+		primaryIndexKey := bytes.Join([][]byte{primaryIndexKeyPrefix, primaryIndexKeySuffix}, nil)
+
 		for i, val := range values {
-			key := encodeColumnKey(cols[i], primaryKey)
+			key := encodeColumnKey(cols[i], primaryIndexKey)
 			if log.V(2) {
-				log.Infof("Put %q -> %v", key, val)
+				log.Infof("CPut %q -> %v", key, val)
 			}
 			// TODO(pmattis): Need to convert the value type to the column type.
 			switch t := val.(type) {
 			case parser.DBool:
-				b.Put(key, bool(t))
+				b.CPut(key, bool(t), nil)
 			case parser.DInt:
-				b.Put(key, int64(t))
+				b.CPut(key, int64(t), nil)
 			case parser.DFloat:
-				b.Put(key, float64(t))
+				b.CPut(key, float64(t), nil)
 			case parser.DString:
-				b.Put(key, string(t))
+				b.CPut(key, string(t), nil)
+			}
```

Comment:
See my comment in @vivekmenezes's change. I think we should structure this as `b.CPut(key, prepareVal(val), nil)`. Fine to do so in a separate PR.